packages/block-editor/src/components/block-list/index.js

```diff
@@ -65,7 +82,9 @@ function Root( { className, ...settings } ) {
 	);
 	return (
 		<elementContext.Provider value={ element }>
-			<div { ...innerBlocksProps } />
+			<IntersectionObserver.Provider value={ intersectionObserver }>
```

Comment:
I've moved the "provider" to "root" because there's no need to have a different intersection observer per "inner blocks" container. One for the whole editor is sufficient.